### ISSUE TITLE
indent treats nested function parameters incorrectly

### ISSUE THREAD

**Issue by seriema** [eslint/eslint#8892]
**Tell us about your environment**

* **ESLint Version:** v4.1.1
* **Node Version:** v5.1.0
* **npm Version:** 3.3.12

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
/*eslint indent: ["error", 2, { "FunctionDeclaration": {"body": 1, "parameters": 1} }]*/
```

</details>

**What did you do? Please include the actual source code causing the issue.**
I ran ESlint to check for validation issues.

```js
/*eslint indent: ["error", 2, { "FunctionDeclaration": {"body": 1, "parameters": 1} }]*/

define([
  "bar",
  "baz",
  "qux"
],

function (
  bar,
  baz,
  qux
) {
  qux();
});
```

**What did you expect to happen?**
I expected no validation issues.

**What actually happened? Please include the actual, raw output from ESLint.**
It gave me validation errors:

```
  10:1  error  Expected indentation of 4 spaces but found 2  indent
  11:1  error  Expected indentation of 4 spaces but found 2  indent
  12:1  error  Expected indentation of 4 spaces but found 2  indent
  13:1  error  Expected indentation of 2 spaces but found 0  indent

✖ 4 problems (4 errors, 0 warnings)
  4 errors, 0 warnings potentially fixable with the `--fix` option.
```

If I run `--fix` it ends up looking like this, which to me is double the function parameter spacing that I want (it has 4 spaces instead of 2) and it adds 2 spaces before `) {`.

```js
define([
  "bar",
  "baz",
  "qux"
],

function (
    bar, // These three parameters should have 2 spaces instead of 4
    baz,
    qux
  ) { // This line should have 0 spaces instead of 2
  qux();
});
```

**Note**
Running without the `define` as a simple method works as expected from the [documentation examples](http://eslint.org/docs/rules/indent#functiondeclaration). I.e. this works as expected (no validation issues):

```js
/*eslint indent: ["error", 2, { "FunctionDeclaration": {"body": 1, "parameters": 1} }]*/

function foo(
  bar,
  baz,
  qux
) {
  qux();
}
```

**Comment by not-an-aardvark** [comment]
Thanks for reporting, I can reproduce this issue.

**Comment by seriema** [comment]
@not-an-aardvark I tried 4.2.0 and the issue isn't really fixed. The example I had above now passes, but if I change the `body` or `parameters` to anything other than 1 it still validates as if it was set to 1.

So changing from
`/*eslint indent: ["error", 2, { "FunctionDeclaration": {"body": 1, "parameters": 1} }]*/`
to
`/*eslint indent: ["error", 2, { "FunctionDeclaration": {"body": 2, "parameters": 2} }]*/` 
or any other values doesn't affect validation.

**Comment by not-an-aardvark** [comment]
I think you want the `FunctionExpression` option, not `FunctionDeclaration`.

If it still isn't working after that, please open a new issue.

**Comment by seriema** [comment]
Oh, I see. Using `FunctionExpression` worked. You can update my bug description if you want, as it didn't work before 4.2.0 but my bug report said `FunctionDeclaration`.
